smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/traits/OperationContextParamsTraitValidator.java

```diff
@@ -0,0 +1,455 @@
+                    events.add(error(operationShape,
+                            String.format("The operation `%s` is marked with `%s` which contains a "
+                                            + "key `%s` with an unparseable JMESPath path `%s`: %s.",
+                                    operationShape.getId(),
+                                    OperationContextParamsTrait.ID.toString(),
+                                    entry.getKey(),
+                                    entry.getValue().getPath(),
+                                    e.getMessage()
+                            )));
+                }
+            }
+        }
+        return events;
+    }
+
+    private LiteralExpression createCurrentNodeFromShape(Shape shape, Model model) {
+        return shape == null
+                ? LiteralExpression.ANY
+                : new LiteralExpression(shape.accept(new ModelRuntimeTypeGenerator(model)));
+    }
+
+    private static final class ModelRuntimeTypeGenerator implements ShapeVisitor<Object> {
+
+        private final Model model;
+        private Set<MemberShape> visited = new HashSet<>();
+
+        ModelRuntimeTypeGenerator(Model model) {
+            this.model = model;
+        }
+
+        @Override
+        public Object blobShape(BlobShape shape) {
+            return "blob";
+        }
+
+        @Override
+        public Object booleanShape(BooleanShape shape) {
+            return true;
+        }
+
+        @Override
+        public Object byteShape(ByteShape shape) {
+            return computeRange(shape);
+        }
+
+        @Override
+        public Object shortShape(ShortShape shape) {
+            return computeRange(shape);
+        }
+
+        @Override
+        public Object integerShape(IntegerShape shape) {
+            return computeRange(shape);
+        }
+
+        @Override
+        public Object longShape(LongShape shape) {
+            return computeRange(shape);
+        }
+
+        @Override
+        public Object floatShape(FloatShape shape) {
+            return computeRange(shape);
+        }
+
+        @Override
+        public Object doubleShape(DoubleShape shape) {
+            return computeRange(shape);
+        }
+
+        @Override
+        public Object bigIntegerShape(BigIntegerShape shape) {
+            return computeRange(shape);
+        }
+
+        @Override
+        public Object bigDecimalShape(BigDecimalShape shape) {
+            return computeRange(shape);
+        }
+
+        @Override
+        public Object documentShape(DocumentShape shape) {
+            return LiteralExpression.ANY;
+        }
+
+        @Override
+        public Object stringShape(StringShape shape) {
+            // Create a random string that does not exceed or go under the length trait.
+            int chars = computeLength(shape);
+
+            // Fill a string with "a"'s up to chars.
+            return new String(new char[chars]).replace("\0", "a");
```

Comment:
This expression is equivalent and more idiomatic
```suggestion
            return "a".repeat(chars);
```

Reply:
I believe `repeat` is only supported in java 11+ :-(   Not sure theres a nicer way in Java 8 that I could find.

Reply:
You're right, I missed that bit. I'm not sure if it's better but we can use `Arrays.fill`, up to you, e.g.,

```
        @Override
        public Object stringShape(StringShape shape) {
            // Create a random string that does not exceed or go under the length trait.
            int chars = computeLength(shape);
            // Fill a string with "a"'s up to chars.
            char[] tmp = new char[chars];
            Arrays.fill(tmp, 'a');
            return new String(tmp);
        }
```